core/db/entities.py

```diff
@@ -109,6 +111,41 @@ class Project(db.Model):
     package = sqla.orm.relationship("Package", back_populates="projects")
 
 
+class ProjectDeliveryPlan(db.Model):
+    """Stores Project Delivery Plan data for projects."""
+
+    __tablename__ = "project_delivery_plan"
+
+    id = sqla.Column(GUID(), default=uuid.uuid4, primary_key=True)  # this should be UUIDType once using Postgres
+    milestone = sqla.Column(sqla.String(), nullable=False)
+    project_id = sqla.Column(sqla.String(), sqla.ForeignKey("project_dim.id"), nullable=False)
+    start_date = sqla.Column(sqla.DateTime(), nullable=False)
+    end_date = sqla.Column(sqla.DateTime(), nullable=False)
+    status = sqla.Column(sqla.Enum(StatusEnum, name="project_delivery_plan_status"), nullable=False)
+
+    # TODO: Should this be nullable? Is Status alone enough in some circumstances?
+    comments = sqla.Column(sqla.String(), nullable=False)
+
+
+class Procurement(db.Model):
+    """Stores Procurement data for projects."""
+
+    __tablename__ = "procurement"
+
+    id = sqla.Column(GUID(), default=uuid.uuid4, primary_key=True)  # this should be UUIDType once using Postgres
+
+    # TODO: Assuming contract is the PK and is unique, only other option I see is that contract + project is unique,
```

Comment:
id is the surrogate PK, and not really relevant, as, by definition it is always unique.

We should add a unique index consisting of contract and project id. Then remove the unique constraint on construction_contract as this would prevent different projects that happen to have the same name from entering them (as per that example data). But unique together in an index means each project cannot have duplicate contract names. Obvs data model people can review though

Reply:
ahh ok makes sense!